src/Utilities/PaymentHelpers.res

```diff
@@ -72,6 +72,72 @@ let retrievePaymentIntent = (clientSecret, headers, ~optLogger, ~switchToCustomP
   })
 }
 
+let threeDsMethod = (url, threeDsMethodData, ~optLogger) => {
+  open Promise
+  logApi(
+    ~optLogger,
+    ~url,
+    ~type_="request",
+    ~eventName=RETRIEVE_CALL_INIT,
+    ~logType=INFO,
+    ~logCategory=API,
+    (),
+  )
+  let threeDsMethodStr = threeDsMethodData->Js.Json.decodeString->Belt.Option.getWithDefault("")
+  let body = `${Js.Global.encodeURIComponent("threeDSMethodData")}=${Js.Global.encodeURIComponent(
```

Comment:
Also please refactor Js.Global

Reply:
Also can we please move these functions outside into variables for better code readability?